src/WebJobs.Script.WebHost/WebScriptHostHttpRoutesManager.cs

```diff
@@ -76,23 +60,46 @@ private void InitializeHttpFunctions(IEnumerable<FunctionDescriptor> functions,
 
                     WebJobsRouteBuilder builder = function.Metadata.IsProxy ? proxiesRoutesBuilder : routesBuilder;
                     builder.MapFunctionRoute(function.Metadata.Name, route, constraints, function.Metadata.Name);
+
+                    LogRouteMap(routesLogBuilder, function.Metadata.Name, route, httpTrigger.Methods, function.Metadata.IsProxy, _httpOptions.Value.RoutePrefix);
                 }
             }
 
             IRouter proxyRouter = null;
             IRouter functionRouter = null;
 
-            if (proxiesRoutesBuilder.Count > 0)
+            if (routesBuilder.Count == 0 && proxiesRoutesBuilder.Count == 0)
             {
-                proxyRouter = proxiesRoutesBuilder.Build();
+                routesLogBuilder.AppendLine("No HTTP routes mapped");
             }
-
-            if (routesBuilder.Count > 0)
+            else
             {
-                functionRouter = routesBuilder.Build();
+                if (proxiesRoutesBuilder.Count > 0)
+                {
+                    proxyRouter = proxiesRoutesBuilder.Build();
+                }
+
+                if (routesBuilder.Count > 0)
+                {
+                    functionRouter = routesBuilder.Build();
+                }
             }
 
             _router.AddFunctionRoutes(functionRouter, proxyRouter);
+
+            ILogger logger = _loggerFactory.CreateLogger("Host.HttpRoutes");
+            logger.LogInformation(routesLogBuilder.ToString());
+        }
+
+        private void LogRouteMap(StringBuilder builder, string functionName, string route, string[] methods, bool isProxy, string prefix)
+        {
+            const string FunctionMappingFormat = "Mapped function route '{4}/{0}' [{1}] to '{2}'{3}";
```

Comment:
this would be easier to read and less error prone if you just had an if/else below and logged the separate messages for each case with a separate interpolated string

Reply:
Went down that path and wasn't happy with the duplication, but I agree that would be less fragile. Will update.